src/submodules/schedulers.jl

```diff
@@ -1,89 +1,101 @@
-export schedule
-export schedule_randomly, schedule_by_id, schedule_fastest, schedule_partially, schedule_by_property, schedule_by_type
-export random_activation, by_id, fastest, partial_activation, property_activation, by_type
+export schedule, Schedulers
 """
-    schedule(model)
+    schedule(model) → ids
 Return an iterator over the scheduled IDs using the model's scheduler.
 """
 schedule(model::ABM) = model.scheduler(model)
 
+# Notice how the above lines are *outside* the submodule
+
+"""
+    Schedulers
+Submodule containing all predefined schedulers of Agents.jl and the scheduling API.
+Schedulers have a very simple interface. They are functions that take as an input the ABM and
+return an iterator over agent IDs. Notice that this iterator can be a "true" iterator
+(non-allocated) or can be just a standard vector of IDs. You can define your own scheduler
+according to this API and use it when making an [`AgentBasedModel`](@ref).
+You can also use the function `schedule(model)` to obtain the scheduled ID list,
+if you prefer to write your own `step!`-like loop.
+
+See also [Advanced scheduling](@ref) for making more advanced schedulers.
+
+Notice that schedulers can be given directly to model creation, and thus become the
+"default" scheduler a model uses, but they can just as easily be incorporated in a
+`model_step!` function as shown in [Advanced stepping](@ref).
+"""
+module Schedulers
+using Agents
```

Comment:
Is this the right way to do this? (more of a TODO myself to look into module design)

Reply:
Answer to this is that parent scope is not accessible in children. We can `import .specific_function` if we want, but I don't think it's all that different.